OverlayPlugin.Core/Controls/WSConfigPanel.cs

```diff
@@ -594,30 +593,16 @@ private bool FetchNgrok(string ngrokPath)
             {
                 UpdateTunnelStatus(TunnelStatus.Downloading);
 
-                simpLogBox.AppendText("Fetching latest ngrok version...\r\n");
-                string dlPage;
-                try
-                {
-                    dlPage = CurlWrapper.Get(NGROK_DOWNLOAD_IDX);
-                } catch (Exception e)
-                {
-                    simpLogBox.AppendText(string.Format("\r\nFailed: {0}\r\n\r\n", e));
-                    return false;
-                }
-
-                var arch = Environment.Is64BitOperatingSystem ? "amd64" : "386";
-                // <a id="windows-dl-link" class="[...]" href="https://bin.equinox.io/c/bNyj1mQVY4c/ngrok-v3-stable-windows-amd64.zip"
-                var match = Regex.Match(dlPage, " href=\"(https://bin.equinox.io/c/[^/]+/ngrok-v3-stable-windows-" + arch + "\\.zip)\"");
-                if (match == Match.Empty)
-                {
-                    simpLogBox.AppendText("Failed to find version on the download page! Please notify ngld or some other dev working on OverlayPlugin.\r\n");
-                    return false;
-                }
+                // Use latest known 2.x because 3.x is incompatible
+                // URLs from https://dl.equinox.io/ngrok/ngrok/stable/archive
+                var ngrokUrl = Environment.Is64BitOperatingSystem ?
+                    "https://bin.equinox.io/a/8exBtGpBr59/ngrok-2.3.40-windows-amd64.zip" :
```

Comment:
How worried are we about minor patches changing the version string and the hash?

Reply:
The archive page contains all previous versions too, so the link to 2.3.40 is unlikely to become unavailable. Development seems focused on the 3.x branch, so I'm not expecting new 2.x versions to appear either, but I could change it to search the archive page for the first `-windows-amd64.zip` link if you think we should.

Reply:
That might break for users again if 3.x versions are ever posted on the same archive page though - so far they haven't been.

Reply:
Gotcha, looks like the archive download page has everything too, so unless 2.3.40 gets removed for a security vulnerability or something it looks like we're okay. Thanks!